pkg/tests/ossm/deploy_on_infra_test.go

```diff
@@ -0,0 +1,183 @@
+		shell.Execute(t,
+			`oc adm taint nodes -l node-role.kubernetes.io/infra node-role.kubernetes.io/infra=reserved:NoSchedule node-role.kubernetes.io/infra=reserved:NoExecute`)
+
+		t.NewSubTest("operator").Run(func(t TestHelper) {
+			t.Log("Verify OSSM Operator is deployed on infra node when configured")
+			t.Log("Reference: https://issues.redhat.com/browse/OSSM-2342")
+			t.Cleanup(func() {
+				shell.Execute(t,
+					`oc patch subscription servicemeshoperator -n openshift-operators --type json -p='[{"op": "remove", "path": "/spec/config/tolerations"}]'`)
+				oc.WaitSMCPReady(t, meshNamespace, smcpName) // Wait for the SMCP to be ready before move to next subtest case (Because untaint)
+			})
+
+			t.LogStep("Patch subscription to run on infra nodes and wait for the operator pod to be ready")
+			oc.Patch(t, "openshift-operators", "subscription", "servicemeshoperator", "merge", `
+spec:
+  config:
+    nodeSelector:
+      node-role.kubernetes.io/infra: ""
+    tolerations:
+    - effect: NoSchedule
+      key: node-role.kubernetes.io/infra
+      value: reserved
+    - effect: NoExecute
+      key: node-role.kubernetes.io/infra
+      value: reserved
+`)
+
+			t.LogStep(fmt.Sprintf("Verify operator pod is running on the infra node. Node expected: %s", workername))
+			retry.UntilSuccess(t, func(t test.TestHelper) {
+				locator := pod.MatchingSelector("name=istio-operator", "openshift-operators")
+				operatorPod := locator(t, oc.DefaultOC)
+				oc.WaitPodReady(t, locator)
+				shell.Execute(t,
+					fmt.Sprintf(`oc get pod -n openshift-operators %s -o jsonpath='{.spec.nodeName}'`, operatorPod.Name),
+					assert.OutputContains(
+						workername,
+						"Success: Operator pod is running on the infra node",
+						"Error: Operator pod is not running on the infra node"))
+			})
+		})
+
+		t.NewSubTest("control plane").Run(func(t TestHelper) {
+			t.Log("Testing: Run OSSM Operator on infra nodes")
+			t.Cleanup(func() {
+				oc.RecreateNamespace(t, meshNamespace)
+				oc.ApplyString(t, meshNamespace, util.RunTemplate(GetSMCPTemplate(env.GetDefaultSMCPVersion()), Smcp))
+				oc.WaitSMCPReady(t, meshNamespace, smcpName)
+			})
+
+			t.LogStep("Patch SMCP to run on infra nodes and wait for the SMCP to be ready")
+			retry.UntilSuccess(t, func(t test.TestHelper) {
+				oc.Patch(t, meshNamespace, "smcp", smcpName, "merge", `
+spec:
+  runtime:
+    defaults:
+      pod:
+        nodeSelector:
+          node-role.kubernetes.io/infra: ""
+        tolerations:
+        - effect: NoSchedule
+          key: node-role.kubernetes.io/infra
+          value: reserved
+        - effect: NoExecute
+          key: node-role.kubernetes.io/infra
+          value: reserved
+`)
+				oc.WaitSMCPReady(t, meshNamespace, smcpName)
+			})
+
+			t.LogStep("Verify that the smcp pods are running on the infra node. Pod expected to be moved: istiod, istio-ingressgateway, istio-egressgateway, jaeger, grafana, prometheus")
+			verifyPodRunningInNode(t)
```

Comment:
The function name is misleading, since it says "Pod" not "Pods".

But I would actually move the for loop out of this function and keep the name "assertPodScheduledToNode" and pass in the pod name and the expected node.